GitUI/UserControls/RevisionGrid.cs

```diff
@@ -427,6 +428,78 @@ private void RevisionsKeyDown(object sender, KeyEventArgs e)
             }
         }
 
+        private void RevisionsKeyUp(object sender, KeyEventArgs e)
+        {
+            if (Revisions.RowCount < LastRowIndex || LastRowIndex < 0 || Revisions.RowCount == 0)
```

Comment:
Can do, though I'm struggling to come up with a good name for this method/property, so open to ideas.
How about this:

```
private int LastSelectedRowIndex 
{
    get 
    {
        if (Revisions.RowCount < LastRowIndex || LastRowIndex < 0 || Revisions.RowCount == 0)
        {
            return -1;
        }
        return LastRowIndex;
    }
}
```

PS: There are lot of improvements can be done in in this class.